### PR TITLE
Add a section documenting uncommon edge cases.

### PR DESCRIPTION
**Pull request by jvlcek**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1540725

### Updates introduced by this PR:

I'm making 2 improvements  with this PR

1. Add *Edge Case* section
2. Add description of one edge case as identified in [BZ 1540725](https://bugzilla.redhat.com/show_bug.cgi?id=1540725)

### To Test:

1. Configure MiQ with authentication _mode: LDAP_  and with an Active Directory host.
1. Specify User Type, of *Distinguished Name (CN=* 
1. Confirm users can successfully log in.
1. Run  miqldap_to_sssd
1. Confirm users can *not* log in.
1. Update file */etc/sssd/sssd.conf* following the instructions introduced by this PR
1. Confirm users can successfully log in.

### CODE REVIEW

**Comment by jvlcek** [comment]
@abellotti and @cybette Please review.

Thank you! JoeV

File: site/_posts/2017-09-29-miqldap-to-sssd.md [hunk 1]
```diff
@@ -342,4 +343,42 @@ This document has diagnostics tips, including:
   - How to use **dbus-send**, with usage examples
   - Adding **debug_level=9** to the sssd.conf to produce more sssd logging information
 
+**Edge Cases**
+=====================================================================
+
+The *miqldap_to_sssd* conversion tool attempts to make assumptions about the initial configuration in order to produce a working SSSD configuration.
+Because there are so many different possible these authentication configurations assumptions may not be completely accurate, resulting in
```
Comment (cybette) [inline site/_posts/2017-09-29-miqldap-to-sssd.md]: different possibilities (?)

File: site/_posts/2017-09-29-miqldap-to-sssd.md [hunk 2]
```diff
@@ -342,4 +343,42 @@ This document has diagnostics tips, including:
...
+an SSSD configuration that my not work correctly.
+
+Updating the *miqldap_to_sssd* conversion tool to handle each edge cases, as they are identified, could result in the  *miqldap_to_sssd* conversion tool
```
Comment (cybette) [inline site/_posts/2017-09-29-miqldap-to-sssd.md]: each edge case

File: site/_posts/2017-09-29-miqldap-to-sssd.md [hunk 3]
```diff
@@ -342,4 +343,42 @@ This document has diagnostics tips, including:
...
+becoming bloated and unwieldy. For this reason it was decided to document edge cases, especially those not likely to be encountered by many users, here.
+
+**Edge Cases 1:** *Active Directory* configured with an atypical *User Type*
```
Comment (cybette) [inline site/_posts/2017-09-29-miqldap-to-sssd.md]: Edge Case 1 (singular)

File: site/_posts/2017-09-29-miqldap-to-sssd.md [hunk 4]
```diff
@@ -342,4 +343,42 @@ This document has diagnostics tips, including:
...
+---------------------------------------------------------------------
+
+**Cause:**
+  1. MiqLdap was configured with an *Active Directory* host.
+  1. An atypical *Active Directory* User Type, of *Distinguished Name (CN=<user>* was specifed.
+  1. When *miqldap_to_sssd* is run it makes the incorrect assumption the that backing directly is *LDAP* because the specified User Type is typical for *LDAP*.
```
Comment (cybette) [inline site/_posts/2017-09-29-miqldap-to-sssd.md]: backing directory (?)

Comment (jvlcek) [reply site/_posts/2017-09-29-miqldap-to-sssd.md]: 👍 Thank you!

File: site/_posts/2017-09-29-miqldap-to-sssd.md (hunk 3 above)
Comment (jvlcek) [reply site/_posts/2017-09-29-miqldap-to-sssd.md]: 👍 Thank you!

File: site/_posts/2017-09-29-miqldap-to-sssd.md (hunk 2 above)
Comment (jvlcek) [reply site/_posts/2017-09-29-miqldap-to-sssd.md]: 👍 Thank you!

**Comment by jvlcek** [comment]
Thank you for the help with this @cybette!

**Comment by cybette** [comment]
@jvlcek my pleasure!

LGTM, @abellotti please take a look as well

File: site/_posts/2017-09-29-miqldap-to-sssd.md [hunk 5]
```diff
@@ -342,4 +343,42 @@ This document has diagnostics tips, including:
   - How to use **dbus-send**, with usage examples
   - Adding **debug_level=9** to the sssd.conf to produce more sssd logging information
 
+**Edge Cases**
+=====================================================================
+
+The *miqldap_to_sssd* conversion tool attempts to make assumptions about the initial configuration in order to produce a working SSSD configuration.
+Because there are so many different possibilities these authentication configurations assumptions may not be completely accurate, resulting in
```
Comment (gtanzillo) [inline site/_posts/2017-09-29-miqldap-to-sssd.md]: I think "configurations" should be "configuration" opn this line

Comment (gtanzillo) [reply site/_posts/2017-09-29-miqldap-to-sssd.md]: Also, I think there should be a comma after "possibilities" on this line

Comment (jvlcek) [reply site/_posts/2017-09-29-miqldap-to-sssd.md]: @gtanzillo Thank you for looking this over. I'm making the suggested changes now.

**Comment by cybette** [comment]
Will merge EOB today if no further comments.

File: site/_posts/2017-09-29-miqldap-to-sssd.md [hunk 6]
```diff
@@ -342,4 +343,42 @@ This document has diagnostics tips, including:
...
+        To: ldap_group_object_class = group
+```
+
+  1. Restart
+
+```console
+     $ systemctl restart sssd.service
+     $ systemctl status sssd.service
+     $ sss_cache -E
+```
```
[the remark this answers is not in the document]
Comment (jvlcek) [reply site/_posts/2017-09-29-miqldap-to-sssd.md]: Good catch. My automated script correctly does it the way you suggest.

```bash
systemctl stop sssd
sss_cache -E
systemctl start sssd
systemctl status sssd
```
